app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  validates_presence_of :username,
+                        :password_digest
```

Comment:
The has_secure_password has this validation built in, and as a bonus will allow a user to update their info without having to give a password during that update. This validation will not allow for that exception so let's pull that, too.

Reply:
I completely forgot about that. Removed it and it still passed the validation test so we're all good on that front now. Thanks for catching that.